v3/src/models/app-state.ts

```diff
@@ -94,13 +102,13 @@ class AppState {
 
   @flow
   *setDocument(
-    snap: ISerializedDocumentModel | ICodapV2DocumentJson,
+    snap: ISerializedV3Document | ICodapV2DocumentJson,
     metadata?: Record<string, any>
   ) {
     // stop monitoring changes for undo/redo on the existing document
     this.disableDocumentMonitoring()
 
-    let content: ISerializedDocumentModel
+    let content: ISerializedV3Document
```

Comment:
Since we might be restoring v2 documents that were saved by v3. The code below that is using `content.revisionId` should probably be changed to `snap.revisionId`. Otherwise I think the conversion from v2 will lose the `revisionId`. That probably also means the type above of `ICodapV2DocumentJson` should be changed to `ISerializedV2Document`